colossus-metrics/src/main/scala/colossus/metrics/Collection.scala

```diff
@@ -117,29 +121,72 @@ class Collection(val config: CollectorConfig) {
         )
       }
     }
-    if (collectors.containsKey(address)) {
-      cast(collectors.get(address))
-    } else {
+    addContext(ns)
```

Comment:
This would only happen if you create a metric context with an address of existing collector, I am not sure how that should be handled. I think we should throw an exception in that case instead of updating tags.